File: Lecture 5 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/subtractProdAndSum.cpp
class Solution {
public:
    int subtractProductAndSum(int n) {
        
        int prod = 1;
        int sum = 0;
        
        while(n!=0) {
            
            int digit = n%10;
            prod = prod * digit;
            sum = sum + digit;
            
            n = n/10;
        }
        
        int answer = prod - sum;
        return answer;
    }
};

File: Lecture005 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/subtractProdAndSum.java
class Solution {
    public int subtractProductAndSum(int n) {

        int product = 1, sum = 0;

        while (n > 0) {
            int digit = n % 10;
            product *= digit;
            sum += digit;
            n = n / 10;
        }

        return product - sum;
    }
}
